cmd/m3fs/main.go

```
@@ -40,6 +40,14 @@ var (
 	clusterDeleteAll bool
 )
 
+// getShaPrefix safely gets the prefix of GitSha and handles the case when GitSha is empty
+func getShaPrefix(sha string) string {
+	if len(sha) >= 7 {
+		return sha[:7]
+	}
+	return "unknown"
```

Don't silently converted the data when `len(sha) > 0 && len(sha) < 7)`